File: test-suite/src/test/java/io/micronaut/docs/function/aws/Handler.java
package io.micronaut.docs.function.aws;

//tag::clazz[]
import io.micronaut.context.env.Environment;
import io.micronaut.core.annotation.Introspected;
import io.micronaut.function.aws.MicronautRequestStreamHandler;

@Introspected
public class Handler extends MicronautRequestStreamHandler {
    @Override
    protected String resolveFunctionName(Environment env) {
        return "eventlogger";
    }
}
//end::clazz[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/function/aws/Handler.kt
package io.micronaut.docs.function.aws
//tag::clazz[]
import io.micronaut.context.env.Environment
import io.micronaut.function.aws.MicronautRequestStreamHandler

class Handler : MicronautRequestStreamHandler() {
    override fun resolveFunctionName(env: Environment): String {
        return "eventlogger"
    }
}
//end::clazz[]